src/uwtools/config/jinja2.py

```diff
@@ -327,32 +328,35 @@ def _report(args: dict) -> None:
     dashes()
 
 
-def _set_up_values_obj(
-    values_file: Optional[Path] = None,
+def _supplement_values(
+    values_src: Optional[Union[dict, Path]] = None,
     values_format: Optional[str] = None,
     overrides: Optional[Dict[str, str]] = None,
+    env: bool = False,
 ) -> dict:
     """
-    Collect template-rendering values based on an input file, if given, or otherwise on the shell
-    environment. Apply override values.
+    Optionally supplement values from given source with overrides and/or environment vairables.
 
-    :param values_file: Path to the file supplying values to render the template.
+    :param values_src: Source of values to render the template.
     :param values_format: Format of values when sourced from file.
-    :param overrides: Supplemental override values.
-    :returns: The collected values.
+    :param overrides: Override values.
+    :param env: Supplement values with environment variables?
+    :returns: The final set of template-rendering values.
     """
-    if values_file:
-        if values_format is None:
-            values_format = get_file_format(values_file)
-        values_class = format_to_config(values_format)
-        values: dict = values_class(values_file).data
-        log.debug("Read initial values from %s", values_file)
+    values: dict
+    if isinstance(values_src, Path):
+        values_format = values_format or get_file_format(values_src)
+        values_src_class = format_to_config(values_format)
+        values = values_src_class(values_src).data
+        log.debug("Read initial values from %s", values_src)
     else:
-        values = dict(os.environ)  # Do not modify os.environ: Make a copy.
-        log.debug("Initial values taken from environment")
+        values = values_src or {}
     if overrides:
         values.update(overrides)
-        log.debug("Updated values with overrides: %s", " ".join(overrides))
+        log.debug("Supplemented values with overrides: %s", " ".join(overrides))
+    if env:
+        values.update(os.environ)
+        log.debug("Supplemented values with environment variables")
```

Comment:
Here's the updated logic to compose `dict`- or file-based values with CLI overrides and/or environment variables. The priority of each source derives from the order of the `update()` calls here.